watchers/chaos_pod_test.go

```diff
@@ -0,0 +1,243 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	"fmt"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/log"
+	"github.com/DataDog/chaos-controller/watchers"
+	"github.com/DataDog/chaos-controller/watchers/mocks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+var _ = Describe("Chaos Pod watcher", func() {
+	Describe("Handler", func() {
+		const bufferSize = 10
+
+		var (
+			chaosPodHandler watchers.ChaosPodHandler
+			disruption      *v1beta1.Disruption
+			eventRecorder   record.EventRecorder
+		)
+
+		JustBeforeEach(func() {
+			// Arrange
+			disruption = &chaosv1beta1.Disruption{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "disruption-name",
+					Namespace: "namespace-name",
+				},
+			}
+			logger, err := log.NewZapLogger()
+			Expect(err).ShouldNot(HaveOccurred())
+			eventRecorder = record.NewFakeRecorder(bufferSize)
+			chaosPodHandler = watchers.NewChaosPodHandler(eventRecorder, disruption, logger)
+		})
+
+		Describe("on Add", func() {
+			var newPod v1.Pod
+
+			JustBeforeEach(func() {
+				// Action
+				chaosPodHandler.OnAdd(&newPod)
+			})
+
+			It("should to nothing", func() {})
```

Comment:
```suggestion
			It("should do nothing", func() {})
```

Reply:
🤔 not sure it worth to have this if there is no expectations associated

Reply:
It just to be explicit about it, do nothing and cover implementation. I practice TDD, so I often defined my unit test before the implementation.